src/AutoMerge/util.jl

```diff
@@ -292,3 +292,34 @@ function get_all_non_jll_package_names(registry_dir::AbstractString)
     unique!(packages)
     return packages
 end
+
+
+"""
+    meets_threshold(threshold, numerator, [denominator])
+
+Test the value of `numerator` or `numerator / denominaror` against
+some threshold to be satisfed.
+
+If the value is an integer then it must be greater than or equal to numerator
+to satisfy the test.
+
+If the value is a float, it must be greater than or equal to ``numerator/denominator`
+to satisfy the test.
+"""
+function meets_threshold end
+
+function meets_threshold(threshold::Integer, linecount::Integer)
+    # @info("meets_threshold", threshold, linecount, linecount >= threshold)
+    linecount >= threshold
+end
+
+function meets_threshold(threshold::Integer, numerator::Integer, ::Integer)
+    # @info("meets_threshold", threshold, numerator, numerator >= threshold)
+    numerator >= threshold
+end
+
+function meets_threshold(threshold::AbstractFloat, numerator::Integer, denominator::Integer)
+    # @info("meets_threshold", threshold, numerator, denominator, (numerator / denominator) >= threshold)
+    (numerator / denominator) >= threshold
+end
```

Comment:
I'm a bit worried about this; I think it's not a good idea to have totally different semantics for integers vs floats, since e.g. `10.0 == 10` but here they would do totally different things. I would suggest starting with only fixed values, and we could add fractions later if we need it. I think just fixed values would likely be enough though.

Reply:
What if I assert that the fliats are between 0 and 1 inclusive?

I feel pretty strongly about fraction support.  10,000 lines of code should not get away with ten lines of documentation or tests.

Reply:
Let's just start with absolute (non-fraction) thresholds, and see how that goes first.

Reply:
Assertions on thresholds are in place.

Reply:
I introduced three new parameters for specifying ratios.  The _min_lines parameters just check absolute line count.  The new _fraction parameters check the ratio of the respective line count to number of source lines or sum of source lines and the respective numerator.